src/libraries/Microsoft.Extensions.Configuration.Binder/gen/TypeSpec.cs

```diff
@@ -7,15 +7,27 @@ namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
 {
     internal record TypeSpec
     {
-        private static SymbolDisplayFormat s_minimalDisplayFormat = new SymbolDisplayFormat(
+        private static readonly SymbolDisplayFormat s_minimalDisplayFormat = new SymbolDisplayFormat(
             globalNamespaceStyle: SymbolDisplayGlobalNamespaceStyle.Omitted,
             typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypes,
-            genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters);
+            genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+            miscellaneousOptions: SymbolDisplayMiscellaneousOptions.UseSpecialTypes);
+
+        private static readonly SymbolDisplayFormat s_specialTypesDisplayFormat = new SymbolDisplayFormat(
+            miscellaneousOptions: SymbolDisplayMiscellaneousOptions.UseSpecialTypes);
 
         public TypeSpec(ITypeSymbol type)
         {
-            FullyQualifiedDisplayString = type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
-            MinimalDisplayString = type.ToDisplayString(s_minimalDisplayFormat);
+            if (type.IsUnmanagedType && type.SpecialType != SpecialType.None)
```

Comment:
@chsienki is there a better check for primitives? I found it odd that this [type](https://github.com/dotnet/runtime/blob/8d7249160b6713c4016850539d55b773a1d100e6/src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.TestClasses.cs#L522-L526) with two containing types (class -> struct -> **struct**) was being reported as unmanaged.

Reply:
No idea @333fred for better advice :)

Reply:
I'm not sure what you're asking for. What do you mean when you say primitive?

Reply:
I mean types that `Type.IsPrimitive` would return true for - https://learn.microsoft.com/en-us/dotnet/api/system.type.isprimitive?view=net-8.0#remarks, along with `string`. My objective here is to display `string` and not `String`, `int` and not `Int32` etc.

Reply:
You don't need anything special for that, `SymbolDisplayFormat.FullyQualifiedFormat` already has `SymbolDisplayMiscellaneousOptions.UseSpecialTypes` set. It will return `string`, not `global::System.String`.

Reply:
Okay thanks I'll take another look at the API.